pkg/apiserver/diagnose/diagnose.go

```diff
@@ -134,7 +134,7 @@ func (s *Service) genReportHandler(c *gin.Context) {
 	db := utils.TakeTiDBConnection(c)
 
 	go func() {
-		defer db.Close()
+		defer utils.RemoveAndCloseTiDBConnFrom(c) //nolint:errcheck
```

Comment:
This might cause problems I guess, since the "db" has already been taken? (That's what "take" means)

Reply:
My fault.